lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -416,6 +416,46 @@ getLocationInfoAsString(const SmallPtrSet<Operation *, 8> &ops) {
   return sstr.str();
 }
 
+/// Given string \p name, generate and return a new name if it conflicts with
+/// any keyword or any other name in the set \p recordNames. Use the int \p
+/// nextGeneratedNameID as a counter for suffix.
+std::string resolveKeywordConflict(const StringRef &origName,
+                                   const llvm::StringSet<> &recordNames,
+                                   size_t &nextGeneratedNameID) {
+  auto name = origName;
+  // Get the list of reserved words we need to avoid.  We could prepopulate this
+  // into the used words cache, but it is large and immutable, so we just query
+  // it when needed.
+  auto &reservedWords = getReservedWords();
+
+  // Check to see if this name is available - if so, use it.
+  if (!reservedWords.count(name))
+    if (recordNames.find(name) == recordNames.end()) {
```

Comment:
Here and below you can use 

`if (!recordNames.count(name))`

Reply:
It would also work a bit better if you change `resolveKeywordConflict` to be responsible for inserting the new name in the recordNames set.  THis would allow you to change this test to be something like:

if (!reservedWords.count(name)) {
 auto itAndInserted = reservedWords.insert(name);
 if (itAndInserted.second)
   return *itAndInserted.first;
}